### ISSUE TITLE
GW2 Crashes quite often

### ISSUE THREAD

**Issue by TheSiiren** [asmaloney/GuildWars2-macOS#9]
### macOS Version

Sonoma 14.2.1

### WineskinServer Version

Wineskin-2.9.2.0

### Wine Engine Version

WS11WineCX64Bit23.7.1_D3DMetal-v1.1

### Problem

Hi Andy,

I recently installed this, the game runs great, but keeps crashing quite often.
I followed all your instructions but used WS11WineCX64Bit23.7.1_D3DMetal-v1.1 and Wineskin-2.9.2.0. Could this have something to do with the crashes?
Hoping you can please help, thanks a million 🙂

**Comment by TheSiiren** [comment]
Oh I should probably add mac specs?
Macbook Pro Apple M1 Max 32 GB memory.

**Comment by asmaloney** [comment]
I don't have an M* mac (I'm still on Intel), but I too have had crashes with recent versions.

The one I'm currently using (Jan 2024) is <strong>WS11WineCX64Bit22.1.1-16</strong>. I'm not sure if there is a D3DMetal version of that one, but if there is, you might try it.

**Comment by TheSiiren** [comment]
There is one. Do you think that might help all the crashes? Do you know what it might be?

**Comment by AamuLumi** [comment]
**TL;DR : update Wineskin `brew install --cask gcenx/wine/wineskin` with the latest version and use WS11Wine64Bit9.3 with Wrapper 3.0 and D3D to Metal translation layer enabled.**

Hey, got some news about the problem.

I've done some tests with a MacBook Pro M1 Max, and I also got some crashes with WS11WineCX64Bit23.7.1_D3DMetal-v1.1, but no one with WS11WineCX64Bit22.1.1-16_D3DMetal-v1.1. The problem is WS11CX22 has poor performance on ARM Mac in comparison of WS11CX23 (which probably have some updates concerning Metal SDK).

I also tested the last wrapper of Wineskin (3.0.0), and got the same crash, so it's not related to the wrapper.

After a crashed test run, I got the exact error which is : 

```
"/Users/gcenx/Documents/GitHub/wineskincx/dlls/ntdll/heap.c: main process heap section" wait timed out in thread 0118, blocked by 014c, retrying (60 sec)
0104:err:sync:RtlpWaitForCriticalSection section 00000000002400D0 "/Users/gcenx/Documents/GitHub/wineskincx/dlls/ntdll/heap.c: main process heap section" wait timed out in thread 0104, blocked by 014c, retrying (60 sec)
00fc:err:sync:RtlpWaitForCriticalSection section 00000000002400D0 "/Users/gcenx/Documents/GitHub/wineskincx/dlls/ntdll/heap.c: main process heap section" wait timed out in thread 00fc, blocked by 014c, retrying (60 sec)
0320:err:sync:RtlpWaitForCriticalSection section 00000000002400D0 "/Users/gcenx/Documents/GitHub/wineskincx/dlls/ntdll/heap.c: main process heap section" wait timed out in thread 0320, blocked by 014c, retrying (60 sec)
034c:err:sync:RtlpWaitForCriticalSection section 00000001427B8F50 (null) wait timed out in thread 034c, blocked by 0320, retrying (60 sec)
0100:err:sync:RtlpWaitForCriticalSection section 00000000002400D0 "/Users/gcenx/Documents/GitHub/wineskincx/dlls/ntdll/heap.c: main process heap section" wait timed out in thread 0100, blocked by 014c, retrying (60 sec)
0108:err:sync:RtlpWaitForCriticalSection section 00000000002400D0 "/Users/gcenx/Documents/GitHub/wineskincx/dlls/ntdll/heap.c: main process heap section" wait timed out in thread 0108, blocked by 014c, retrying (60 sec)
```

I checked the Wine issue tracker, and found this in few games. I tested a solution based on install `xact` with `winetricks`, but it doesn't change anything.

I saw Wineskin main dev' release a new [Wineskin version](https://github.com/Gcenx/WineskinServer/issues/530#issuecomment-1998418720) yesterday, and this version include two important things : 
- new Wine engines (version 9.1/9.2/9.3)
- a toggle to enable Direct3D to Metal translation layer.

I tested with WS11Wine64Bit9.3, Wineskin-3.0 and setting for D3D to Metal translation layer (in Configuration tab of the GW2.app), and I had very good performance and no crash after 4 or 5 hours playing GW2.
During these hours, I tested the game essentially in WvW following a group of 40 players (which was my normally crash case), and the only problem I got was some occasional freezes during 300/500ms (I believe it was these freezes which became crashes with Wine 8).

So in any case someone wants to play GW2 on ARM Mac, here's a screenshot of my working config : 

<img width="676" alt="Capture d’écran 2024-03-15 à 21 26 20" src="https://github.com/asmaloney/GuildWars2-macOS/assets/1839713/9382a3e6-ffc9-4809-903e-c7f420d9c04f">

**Comment by TheSiiren** [comment]
WOW! Thank you for all your help and research on this. I really appreciate it!
Let me give it a run.

Does wineskin work for other Windows games as well?

**Comment by AamuLumi** [comment]
You're welcome !

> Does wineskin work for other Windows games as well?

I saw issues on Wine tracker and Wineskin tracker relative to a lot of other games, so it should work for other games.
For example, I know the Final Fantasy XIV unofficial MacOS client is based on a similar stack and I never had problem with it.

Here's the Wineskin repo if you want to take a look : https://github.com/Gcenx/WineskinServer

**Comment by TheSiiren** [comment]
Would I be able to use it for Valorant?

**Comment by AamuLumi** [comment]
> Would I be able to use it for Valorant?

Probably not, because Valorant uses the Riot custom anti-cheat Vanguard, and anti-cheats don't work well with Wine and other Windows emulation software.

**Comment by TheSiiren** [comment]
https://blix.gg/news/how-to-install-valorant-on-mac-m1m2

I found this, do you think it would work?

**Comment by asmaloney** [comment]
(Closing as it seems like the original issue is addressed and it's moving on to other topics not related to this repo - hope you find a solution!)
